tests/integration/test_integration.py

```diff
@@ -635,16 +635,22 @@ def test_area_management(client: TestClient, admin_access_token: str) -> None:
     # Ensure the layer is deleted
     res = client.get(f"/v1/studies/{study_id}/layers")
     assert res.json() == [
-        LayerInfoDTO(id="0", name="All", areas=["area 1", "area 2"]).model_dump(),
+        Layer(id="0", name="All", areas=["area 1", "area 2"]).model_dump(),
     ]
 
     # Try to delete a non-existing layer
     res = client.delete(f"/v1/studies/{study_id}/layers/1")
-    assert res.status_code == HTTPStatus.NOT_FOUND
+    assert res.json() == {
```

Comment:
Indeed in other "managers", we duplicate controls before applying the command in order to get that kind of proper status code, not a 500, which is not even raised for variants since the command will not be applied instantly.

The duplication is not great and we could probably find a way to reduce it, for example by having a valdiation method in commands, but I think we should keep the behaviour.